### PR TITLE
chore(gatsby): Convert get-ssl-cert to typescript

### PR DESCRIPTION
**Pull request by mottox2**
## Description

Convert get-ssl-cert to typescript

## Related Issues

related to #21995

### CODE REVIEW

File: packages/gatsby/src/commands/develop.ts [hunk 1]
```diff
@@ -401,12 +401,12 @@ module.exports = async (program: IProgram): Promise<void> => {
...
     }
 
-    program.ssl = await getSslCert({
+    program.ssl = (await getSslCert({
       name: sslHost,
       certFile: program[`cert-file`],
       keyFile: program[`key-file`],
       directory: program.directory,
-    })
+    })) as ICert
```
Comment (pvdz) [inline packages/gatsby/src/commands/develop.ts:409]: I don't think this should be casted (surely TS can unbox the promise and type properly with a syntax `await`?). 

Additionally, the return value can be `false` here so the cast seems insufficient to me.

File: packages/gatsby/src/utils/get-ssl-cert.ts [hunk 2]
```diff
@@ -25,11 +38,11 @@ module.exports = async ({ name, certFile, keyFile, directory }) => {
     const keyPath = absoluteOrDirectory(directory, keyFile)
     const certPath = absoluteOrDirectory(directory, certFile)
 
-    return await {
```
Comment (pvdz) [inline packages/gatsby/src/utils/get-ssl-cert.ts:28]: heh. thanks!

File: packages/gatsby/src/utils/get-ssl-cert.ts [hunk 3]
```diff
@@ -1,17 +1,30 @@
...
 
-module.exports = async ({ name, certFile, keyFile, directory }) => {
+interface IGetSslCertArgs {
+  name: string
+  certFile?: string
+  keyFile?: string
+  directory: string
+}
+
+export const getSslCert = async ({
```
Comment (pvdz) [inline packages/gatsby/src/utils/get-ssl-cert.ts:22]: If we're changing this, can you change it to a function?

```suggestion
export async function getSslCert({
```

File: packages/gatsby/src/utils/__tests__/get-ssl-cert.ts [hunk 4]
```diff
@@ -28,9 +28,9 @@ describe(`gets ssl certs`, () => {
     certificateFor.mockClear()
   })
   describe(`Custom SSL certificate`, () => {
-    it.each([[{ certFile: `foo` }], [{ keyFile: `bar` }]])(
+    it.each([{ certFile: `foo` }, { keyFile: `bar` }])(
       `panic if cert and key are not both included`,
-      args => {
+      (args: any): void => {
```
Comment (pvdz) [inline packages/gatsby/src/utils/__tests__/get-ssl-cert.ts:33]: Why not cast this explicitly to `IGetSslCertArgs`?

**Comment by blainekasten** [comment]
@mottox2 This PR is very outdated now :( Sorry we couldn't move faster on it. Would you be able to resolve the conflicts and we'll get it merged?

File: packages/gatsby/src/commands/develop.ts (hunk 1 above)
Comment (blainekasten) [reply packages/gatsby/src/commands/develop.ts]: fixed

File: packages/gatsby/src/utils/get-ssl-cert.ts (hunk 3 above)
Comment (blainekasten) [reply packages/gatsby/src/utils/get-ssl-cert.ts:45]: fixed

File: packages/gatsby/src/utils/__tests__/get-ssl-cert.ts (hunk 4 above)
Comment (blainekasten) [reply packages/gatsby/src/utils/__tests__/get-ssl-cert.ts:33]: fixed

**Pull request merged**